app/results.py

```diff
@@ -52,17 +52,23 @@ def is_possible_phone_number(query_str):
 
 
 class Handler(BaseHandler):
-    def search(self, query):
+    def search(self, query_txt):
         """Performs a search and adds view_url attributes to the results."""
         results = None
         if self.config.external_search_backends:
             results = external_search.search(
-                self.repo, query, MAX_RESULTS,
+                self.repo, TextQuery(query_txt), MAX_RESULTS,
                 self.config.external_search_backends)
+
         # External search backends are not always complete. Fall back to the
         # original search when they fail or return no results.
         if not results:
-            results = indexing.search(self.repo, query, MAX_RESULTS)
+            if self.config.enable_fulltext_search:
```

Comment:
Can you use config.get('enable_fulltext_search') instead? self.config.enable_fulltext_search supports repository-specific config but config.get('enable_fulltext_search') doesn't. So mixing these two can cause confusing behavior.